core/store/modules/product/actions.ts

```diff
@@ -44,36 +44,43 @@ const actions: ActionTree<ProductState, RootState> = {
       }
       context.state.breadcrumbs.routes = breadCrumbRoutes(path) // TODO: change to store.commit call?
     }
-    // TODO: Fix it when product is enterd from outside the category page
-    let currentPath = context.rootState.category.current_path
-    let currentCat = context.rootState.category.current
 
-    if (currentPath.length > 0 && currentCat) {
-      setbrcmb(currentPath)
-    } else {
-      if (product.category && product.category.length > 0) {
-        subloaders.push(
-          context.dispatch('category/list', {}, { root: true }).then((categories) => {
-            for (let cat of product.category.reverse()) {
-              let category = categories.items.find((itm) => { return itm['id'] === cat.category_id })
-              if (category) {
-                context.dispatch('category/single', { key: 'id', value: category.id }, { root: true }).then((category) => { // this sets up category path and current category
-                  setbrcmb(context.rootState.category.current_path)
-                }).catch(err => {
-                  setbrcmb(context.rootState.category.current_path)
-                  console.error(err)
-                })
-                break
-              }
+    if (product.category && product.category.length > 0) {
+      const categoryIds = product.category.reverse().map((cat => cat.category_id))
+
+      subloaders.push(
+        context.dispatch('category/list', {}, { root: true }).then((categories) => {
+          const catList = []
+
+          for (let catId of categoryIds) {
+            let category = categories.items.find((itm) => { return itm['id'] === parseInt(catId) })
+            if (category) {
+              catList.push(category)
             }
+          }
+
+          const rootCat = catList.shift()
+          let catForBreadcrumbs = rootCat
+
+          for (let cat of catList) {
+            const catPath = cat.path
+            if (catPath && catPath.includes(rootCat.path) && (catPath.split('/').length > catForBreadcrumbs.path.split('/').length)) {
```

Comment:
The path field was not required before and if somebody is using for example M1 connector can have problem with this field being empty.

Can we easily add a fallback to the previous behavior - modified just by the way you’re selecting the root category? 

So when the path is not set let’s generate breadcrumbs the previous way with parents